Jenkinsfile

```diff
@@ -16,119 +22,184 @@ pipeline {
   environment {
     REPO = 'sw_usb_audio'
     VIEW = getViewName(REPO)
+    TOOLS_VERSION = "15.2.1"
+    XTAGCTL_VERSION = "v2.0.0"
   }
   stages {
     stage('Build') {
-      agent {
-        label 'linux && x86_64'
-      }
-      stages {
-        stage('Get view') {
+      parallel {
+        stage('Build applications') {
+          // Use XCommon CMake to fetch dependencies, but then build using legacy XCommon Makefiles
+          agent {
+           label 'linux && x86_64'
+          }
           steps {
-            xcorePrepareSandbox("${VIEW}", "${REPO}")
+            println "Stage running on ${env.NODE_NAME}"
+            get_xcommon_cmake()
+
+            dir("${REPO}") {
+              checkout scm
+
+              withTools("${env.TOOLS_VERSION}") {
+                withEnv(["XMOS_CMAKE_PATH=${WORKSPACE}/xcommon_cmake"]) {
+                  // Fetch all dependencies using XCommon CMake
+                  sh "cmake -G 'Unix Makefiles' -B build"
+
```

Comment:
Could we use cmake --build here to ensure every builds (unsure how we specifiy using xmake though)

Reply:
For the XCommon CMake build, I'm already doing this by running cmake at the top-level and then just xmake in the build directory to build everything. It's a pain to do the equivalent for the legacy xmake build because for some reason it switches to single threaded mode when using the top-level Makefile, so the build of all the configs across all applications will take far too long for our regression job.